MdePkg/Include/AArch64/AArch64.h

```diff
@@ -22,7 +22,8 @@
 #define CPACR_CP_FULL_ACCESS  0x300000
 
 // Coprocessor Trap Register (CPTR)
-#define AARCH64_CPTR_TFP  (1 << 10)
+#define AARCH64_CPTR_TFP             (1 << 10)
+#define AARCH64_CPTR_CP_FULL_ACCESS  0x33FF

```

Comment:
God, this line is full of technical debt. Past, not new.

So, the CPTR *used* to be the Coprocessor Trap Register, since it emulates something that's been around since the 32-bit Arm architecture had an actual coprocessor expansion bus, which had its encoding space eventually reused for new and shiny architectural features (like Advanced SIMD).

The fully written out name of this register in every version of the 64-bit Arm Arm since the first released one is however "Architectural Feature Trap Register" ... even though the mnemonic stayed the same. So, I think it's somewhat incorrect to inherit the _CP_ aspect for this macro.

Now, apart from that - the access control register was just that: an access register.
This is a trap register, so the logic is inverted. So I don't think FULL_ACCESS is semantically correct either.
The value 0x33FF means FEAT_SME instructions are trapped, Advanced SIMD and floating-point instructions are not trapped, FEAT_SVE instructions are trapped. That probably *is* what we want, and any other solution requires explicitly checking for feature support inline.

But maybe the macro should be called something like AARCH64_CPTR_FP_UNTRAPPED?

Reply:
The "CP_FULL_ACCESS" is copied from CPACR where it's equally poorly named (and I noted that when copying it, but consistency and all that...). Yes, the meanings of some fields are inverted and sometimes more than one bit, but broadly speaking these two bit patterns do the same thing at their respective ELs, trapping everything except SIMD/FP, which is hardly "full access". Of course, the CP part of it is inherited from AArch64's poor naming of the thing, but it's also a bit redundant with the register name prefix...